File: common/protcol/js/caller/hub_call_dbproxycaller.js
/*this caller file is codegen by juggle for js*/
function hub_call_dbproxy_caller(ch){
    Icaller.call(this, "hub_call_dbproxy", ch);

    this.reg_hub = function( argv0){
        var _argv = [argv0];
        this.call_module_method.call(this, "reg_hub", _argv);
    }

    this.create_persisted_object = function( argv0, argv1, argv2, argv3){
        var _argv = [argv0,argv1,argv2,argv3];
        this.call_module_method.call(this, "create_persisted_object", _argv);
    }

    this.updata_persisted_object = function( argv0, argv1, argv2, argv3, argv4){
        var _argv = [argv0,argv1,argv2,argv3,argv4];
        this.call_module_method.call(this, "updata_persisted_object", _argv);
    }

    this.get_object_count = function( argv0, argv1, argv2, argv3){
        var _argv = [argv0,argv1,argv2,argv3];
        this.call_module_method.call(this, "get_object_count", _argv);
    }

    this.get_object_info = function( argv0, argv1, argv2, argv3){
        var _argv = [argv0,argv1,argv2,argv3];
        this.call_module_method.call(this, "get_object_info", _argv);
    }

    this.remove_object = function( argv0, argv1, argv2, argv3){
        var _argv = [argv0,argv1,argv2,argv3];
        this.call_module_method.call(this, "remove_object", _argv);
    }

}
(function(){
    var Super = function(){};
    Super.prototype = Icaller.prototype;
    hub_call_dbproxy_caller.prototype = new Super();
})();
hub_call_dbproxy_caller.prototype.constructor = hub_call_dbproxy_caller;

File: protcol/c#/caller/hub_call_dbproxycaller.cs
/*this caller file is codegen by juggle for c#*/
using System;
using System.Collections;
using System.IO;

namespace caller
{
    public class hub_call_dbproxy : juggle.Icaller 
    {
        public hub_call_dbproxy(juggle.Ichannel _ch) : base(_ch)
        {
            module_name = "hub_call_dbproxy";
        }

        public void reg_hub(String argv0)
        {
            ArrayList _argv = new ArrayList();
            _argv.Add(argv0);
            call_module_method("reg_hub", _argv);
        }

        public void create_persisted_object(String argv0,String argv1,Hashtable argv2,String argv3)
        {
            ArrayList _argv = new ArrayList();
            _argv.Add(argv0);
            _argv.Add(argv1);
            _argv.Add(argv2);
            _argv.Add(argv3);
            call_module_method("create_persisted_object", _argv);
        }

        public void updata_persisted_object(String argv0,String argv1,Hashtable argv2,Hashtable argv3,String argv4)
        {
            ArrayList _argv = new ArrayList();
            _argv.Add(argv0);
            _argv.Add(argv1);
            _argv.Add(argv2);
            _argv.Add(argv3);
            _argv.Add(argv4);
            call_module_method("updata_persisted_object", _argv);
        }

        public void get_object_count(String argv0,String argv1,Hashtable argv2,String argv3)
        {
            ArrayList _argv = new ArrayList();
            _argv.Add(argv0);
            _argv.Add(argv1);
            _argv.Add(argv2);
            _argv.Add(argv3);
            call_module_method("get_object_count", _argv);
        }

        public void get_object_info(String argv0,String argv1,Hashtable argv2,String argv3)
        {
            ArrayList _argv = new ArrayList();
            _argv.Add(argv0);
            _argv.Add(argv1);
            _argv.Add(argv2);
            _argv.Add(argv3);
            call_module_method("get_object_info", _argv);
        }

        public void remove_object(String argv0,String argv1,Hashtable argv2,String argv3)
        {
            ArrayList _argv = new ArrayList();
            _argv.Add(argv0);
            _argv.Add(argv1);
            _argv.Add(argv2);
            _argv.Add(argv3);
            call_module_method("remove_object", _argv);
        }

    }
}
